### PR TITLE
doc: 6908 add tarball verification docs v1

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6908

Describe changes:
- add Suricata tarball verification documentation
- RTD build https://rtd-test-suricata.readthedocs.io/en/latest/source-verification.html
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: doc/userguide/install.rst [hunk 1]
```diff
@@ -16,6 +16,9 @@ Source
 
 Installing from the source distribution files gives the most control over the Suricata installation.
 
+The Suricata source tarballs should be verified before building the source, see
+:doc:`source-verification`
```
Comment (jasonish) [inline doc/userguide/install.rst:20]: Can we use common naming here?  "source distribution" or "source tarball"?

File: doc/userguide/source-verification.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+------------------
+
+These verification steps are for general guidance, the exact process and
+commands may vary between operating systems.
+
+Downloading the sig File
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The sig file needs to be downloaded as well as the tarball. Both files can be
+found at `<https://suricata.io/download/>`_.
```
Comment (jasonish) [inline doc/userguide/source-verification.rst:21]: "signature" is how refer to it on the download page, should match I think.

Comment (jmtaylor90) [reply doc/userguide/source-verification.rst:21]: will update in next PR

File: doc/userguide/install.rst (hunk 1 above)
Comment (jmtaylor90) [reply doc/userguide/install.rst:20]: Ah I hadn't paid attention to the previous lines, will update to match in the next PR. Any preference on wording?

Comment (jasonish) [reply doc/userguide/install.rst:20]: Not a big fan of "tarball" in official documentation.  I prefer "archive" or "distribution archive", etc.

Comment (jmtaylor90) [reply doc/userguide/install.rst:20]: Alright will get that updated/reworked, thanks!

Comment (jufajardini) [reply doc/userguide/install.rst:20]: nit: missing the period.

File: doc/userguide/source-verification.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
+
```
Comment (jufajardini) [inline doc/userguide/source-verification.rst:1]: nit: remove empty line

**Review by jufajardini — CHANGES_REQUESTED** [review]
Some of the errors from the CI are doc building or dist check complaints.
```
/__w/suricata/suricata/doc/userguide/index.rst:6:toctree contains reference to nonexisting document 'source-verification'
```
(e.g. https://github.com/OISF/suricata/actions/runs/9019965517/job/24784406162?pr=11051#step:11:1108)

My only initial guess is that maybe that empty line before the chapter title makes some distros unhappy? Or there's something else I can't see.

**Comment by jufajardini** [comment]
The other failures are related to: https://github.com/OISF/suricata/pull/11041

File: doc/userguide/source-verification.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Downloading the sig File
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The sig file needs to be downloaded as well as the tarball. Both files can be
+found at `<https://suricata.io/download/>`_.
+
+Importing the OISF Signing Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once both the sig file and suricata tarball files are obtained, the OISF
```
Comment (jufajardini) [inline doc/userguide/source-verification.rst:26]: nit: consistent usage of **S**uricata, please? :)

File: doc/userguide/source-verification.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  gpg: Total number processed: 1
+  gpg:               imported: 1
+
+Verifying the Suricata Tarball
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To verify the contents of the Suricata tarball the following command should be
+ran::
+
+  $ gpg --verify suricata-7.0.5.tar.gz.sig suricata-7.0.5.tar.gz
```
Comment (jufajardini) [inline doc/userguide/source-verification.rst:45]: Considering how literal some interpretations can be, I think it's worth mentioning that the proper Suricata version should match with what they have installed, in their command-run. 

Or maybe rephrase to indicate "To verify the Suricata-7.0.5 distribution archive, for instance, you could run"

File: doc/userguide/source-verification.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    gpg:                using RSA key B36FDAF2607E10E8FFA89E5E2BA9C98CCDF1E93A
+    gpg: Good signature from "Open Information Security Foundation (OISF)
+    <releases@openinfosecfoundation.org>" [unknown]
+    gpg: WARNING: This key is not certified with a trusted signature!
+    gpg:          There is no indication that the signature belongs to the owner.
+    Primary key fingerprint: B36F DAF2 607E 10E8 FFA8  9E5E 2BA9 C98C CDF1 E93A
+
+  This indicates that the OISF signing key was imported and the signatures are
+  valid, but either the keys have not been marked as trusted OR the keys are
+  possibly a forgery.
```
Comment (jufajardini) [inline doc/userguide/source-verification.rst:83]: Maybe add what could be done in this case? Don't trust the package, or reach out to us to let us know? Which would be the best channel for that? info@oisf.net?

Comment (jmtaylor90) [reply doc/userguide/source-verification.rst:83]: Good question, what about something like "If there are questions regarding the validity of the downloaded file, the OISF team can be reached via any of the methods at https://suricata.io/our-story/contact/" ?

Comment (jufajardini) [reply doc/userguide/source-verification.rst:83]: I think it's best if we point to something specific, to reduce friction. Not sure if this would be considered a security issue. If so, they could write to security@oisf.net (cf  https://github.com/OISF/suricata/blob/master/SECURITY.md). If not, maybe the contact e-mail: info@oisf.net

**Comment by jmtaylor90** [comment]
continued in https://github.com/OISF/suricata/pull/11111

**Comment by jufajardini** [comment]
> Some of the errors from the CI are doc building or dist check complaints.
> 
> ```
> /__w/suricata/suricata/doc/userguide/index.rst:6:toctree contains reference to nonexisting document 'source-verification'
> ```
> 
> (e.g. https://github.com/OISF/suricata/actions/runs/9019965517/job/24784406162?pr=11051#step:11:1108)
> 
> My only initial guess is that maybe that empty line before the chapter title makes some distros unhappy? Or there's something else I can't see.

If anyone by any change arrives here, the solution was actually to add the new file to the Makefile.am, as suggested by Jeff (cf https://github.com/OISF/suricata/pull/11111#issuecomment-2127123594)
